internal/api/external.go

```diff
@@ -203,9 +202,12 @@ func (a *API) internalExternalProviderCallback(w http.ResponseWriter, r *http.Re
 	// if there's a non-empty FlowStateID we perform PKCE Flow
 	if flowStateID := getFlowStateID(ctx); flowStateID != "" {
 		flowState, err = models.FindFlowStateByID(a.db, flowStateID)
-		if err != nil {
-			return err
+		if models.IsNotFoundError(err) {
+			return unprocessableEntityError(ErrorCodeFlowStateNotFound, "Flow state not found").WithInternalError(err)
```

Comment:
Should we return a `notFoundError` here instead? 

Though, I guess this would also fit 422 since it's structurally correct but fails validation rules.

Reply:
Ah looks like we do the same here https://github.com/supabase/gotrue/pull/1377/files#diff-69644f329a0107a4e9977ffbf1aa73996cdc789d1cd073a237732d2ab2a956adR523 so I'm guessing this is intended